### PR TITLE
Issue #181 - Consolidate 'add-chain' and 'add-pre-chain' endpoints

### PR DESCRIPTION
**Pull request by bifurcation**

### CODE REVIEW

File: draft-ietf-trans-rfc6962-bis.md [hunk 1]
```diff
@@ -1028,11 +1028,16 @@ POST https://\<log server>/ct/v2/add-chain
...
-    certificate for which the submitter desires an SCT; the second certifies the
-    first and so on to the last, which either is, or is certified by, an
-    accepted trust anchor.
+  : An array of base64 encoded certificates. If the certificate is being
+    submitted as a precertificate, then the first element is the signer of the
+    precertificate; otherwise, the first element is the certificate being
+    submitted.  The second certifies the first and so on to the last, which
+    either is, or is certified by, an accepted trust anchor.
 
 Outputs:
```
Comment (eranmes) [inline draft-ietf-trans-rfc6962-bis.md]: To not "break" the document while we move the SCT type into the SCT data structure itself, the outputs section for add-chain should indicate that the type of the returned `TransItem` is either x509_sct_v2 OR precert_sct_v2

File: draft-ietf-trans-rfc6962-bis.md [hunk 2]
```diff
@@ -1028,11 +1028,16 @@ POST https://\<log server>/ct/v2/add-chain
 
 Inputs:
 
+: precertificate:
+  : The base64 encoded precertificate.  This field MUST be included if and only
+    if the certificate is being submitted as a precertificate.
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: In RFC6962, precertificates are X.509 certificates.  However, in 6962-bis, a precertificate is a CMS object containing a TBSCertificate.  So "certificate is being submitted as a precertificate" doesn't make sense.

How about shortening the description to ": The base64 encoded precertificate, if applicable." ?

File: draft-ietf-trans-rfc6962-bis.md [hunk 3]
```diff
@@ -1028,11 +1028,16 @@ POST https://\<log server>/ct/v2/add-chain
...
 : chain:
-  : An array of base64 encoded certificates. The first element is the
-    certificate for which the submitter desires an SCT; the second certifies the
-    first and so on to the last, which either is, or is certified by, an
-    accepted trust anchor.
+  : An array of base64 encoded certificates. If the certificate is being
+    submitted as a precertificate, then the first element is the signer of the
+    precertificate; otherwise, the first element is the certificate being
+    submitted.  The second certifies the first and so on to the last, which
+    either is, or is certified by, an accepted trust anchor.
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: "the certificate being submitted" doesn't make sense either, because it's actually multiple certificates (i.e., a certificate chain) that are being submitted.

I'd reword the description of the `chain` input as follows:
"An array of base64 encoded certificates. If the `precertificate` input is specified, then the first element is the signer of the precertificate; otherwise, the first element is the certificate for which the submitter desires an SCT. The second element certifies the first and so on to the last, which either is, or is certified by, an accepted trust anchor."

File: draft-ietf-trans-rfc6962-bis.md [hunk 4]
```diff
@@ -503,7 +503,7 @@ submit them.
 ## Precertificates    {#precertificates}
 
 CAs may preannounce a certificate prior to issuance by submitting a
-precertificate ({{add-pre-chain}}) that the log can use to create an entry that
+precertificate ({{add-chain}}) that the log can use to create an entry that
```
Comment (robstradling) [inline draft-ietf-trans-rfc6962-bis.md]: I think we should change the name of the consolidated endpoint to "add-entry".

**Comment by eranmes** [comment]
Closed in favour of https://github.com/google/certificate-transparency-rfcs/pull/248.
